core.cc

```diff
@@ -0,0 +1,152 @@
+#include <node.h>
+#include <nan.h>
+#include <syslog.h>
+
+using v8::Function;
+using v8::FunctionTemplate;
+using v8::Handle;
+using v8::Local;
+using v8::Null;
+using v8::Number;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+class Worker : public NanAsyncWorker {
+    public:
+        Worker(NanCallback *callback, int priority, Handle<String> message_)
+            : NanAsyncWorker(callback), priority(priority), message(0) {
+                message = new char[message_->Length() * 4];
+                message_->WriteUtf8(message);
+            }
+        ~Worker() {
+            delete[] message;
+        }
+
+        void Execute () {
+            syslog(priority, "%s", message);
+        }
+
+        void HandleOKCallback () {
+            NanScope();
+
+            if(!callback->IsEmpty())
+                callback->Call(0, 0);
+        };
+
+    private:
+        int priority;
+        char* message;
+};
+
+// XXX(sam) put in anonymous namespace {
+static char ident[1024];
+
+// void openlog(const char *ident, int option, int facility);
+NAN_METHOD(OpenLog) {
+    NanScope();
+
+    // openlog requires ident be statically allocated. Write doesn't guarantee
+    // NULL-termination, so preserve last byte as NULL.
+    args[0]->ToString()->WriteUtf8(ident, sizeof(ident)-1);
+    int option = args[1]->ToInt32()->Value();
+    int facility = args[2]->ToInt32()->Value();
+
+    openlog(ident, option, facility);
+
+    NanReturnUndefined();
+}
+
+// void syslog(int priority, const char *format, ...);
+NAN_METHOD(SysLog) {
+    NanScope();
+
+    int priority = args[0]->ToInt32()->Value();
+    Handle<String> message = args[1]->ToString();
+    NanCallback *callback = new NanCallback(args[2].As<Function>());
+
+    NanAsyncQueueWorker(new Worker(callback, priority, message));
+
+    NanReturnUndefined();
+}
+
+// int setlogmask(int mask);
+NAN_METHOD(SetLogMask) {
+    NanScope();
+
+    int mask = args[0]->ToInt32()->Value();
+    int last = setlogmask(mask);
+
+    NanReturnValue(NanNew<Number>(last));
+}
+
+// void closelog(void);
+NAN_METHOD(CloseLog) {
+    NanScope();
+
+    closelog();
+
+    NanReturnUndefined();
+}
+
+void Init(Handle<Object> exports) {
+    // XXX(sam) does String have a .ToLower()?
```

Comment:
It does not.  You could look up `String.prototype.toLowerCase` through the context's global object and invoke that but meh.